src/main/java/org/ehcache/sizeof/ObjectGraphWalker.java

```diff
@@ -263,6 +263,14 @@ private static void nullSafeAdd(final Deque<Object> toVisit, final Object o) {
                         LOG.error("Security settings prevent Ehcache from accessing the subgraph beneath '{}'" +
                                   " - cache sizes may be underestimated as a result", field, e);
                         continue;
+                    } catch (RuntimeException e) {
+                        // new class (jdk9+ only) can be thrown
+                        if ("java.lang.reflect.InaccessibleObjectException".equals(e.getClass().getName())) {
+                            LOG.error("JPMS blocks field access. This prevents Ehcache from accessing "
```

Comment:
This should probably be just a warning. Because it might occur a lot and there's nothing we can do about it.